binder.go

```diff
@@ -26,6 +27,11 @@ func (binder *NoUnknownFieldsBinder) Bind(i interface{}, c echo.Context) error {
 	// no fields were plucked from the body.
 	isInitiallyEmpty := reflect.ValueOf(i).Elem().IsZero()
 
+	// Check if request contains a body
+	if c.Request().Body == http.NoBody {
```

Comment:
Can we check if the body is `nil` here too? I think we were checking for both before because iirc a `net/http` request can have either of those options when there isn't a body associated with the request. 

Also can we move this check to the top - no need to run the reflection checks if there isn't a body.

Reply:
ok, I will change it ... but I dont know how to test the "nil" value here because everything I tried lead to http.NoBody ... any idea? 🤔

Reply:
Turns out I misread the docs, we don't need the check: https://pkg.go.dev/net/http#Request.Body

The first part specifies that a *client* request can have a nil body, whereas a server request will always have a non-nil body (e.g. one that returns `EOF` and/or is `http.NoBody`).